fastp_0.20.1/fastp_0.20.1.cwl

```diff
@@ -0,0 +1,130 @@
+class: CommandLineTool
+cwlVersion: v1.0
+$namespaces:
+  dct: 'http://purl.org/dc/terms/'
+  doap: 'http://usefulinc.com/ns/doap#'
+  foaf: 'http://xmlns.com/foaf/0.1/'
+  sbg: 'https://www.sevenbridges.com/'
+id: fastp_0_20_1
+baseCommand:
+  - fastp
+inputs:
+  - id: read1_input
+    type: File
+    inputBinding:
+      prefix: '--in1'
+    doc: |
+        read1 input file name
+  - id: read1_output_path
+    type: string
+    inputBinding:
+      prefix: '--out1'
+    doc: |
+        read1 output file name
+  - id: read2_input
+    type: File?
+    inputBinding:
+      prefix: '--in2'
+    doc: |
+        read2 input file name, for PE data
+  - id: read2_output_path
+    type: string?
+    inputBinding:
+      prefix: '--out2'
+    doc: |
+        read2 output file name
+  - id: unpaired1_path
+    type: string?
+    inputBinding:
+      prefix: '--unpaired1'
+    doc: |
+        for PE input, if read1 passed QC but read2 not, it will be written to unpaired1. 
+  - id: unpaired2_path
+    type: string?
+    inputBinding:
+      prefix: '--unpaired2'
+    doc: |
+        for PE input, if read2 passed QC but read1 not, it will be written to unpaired2.
+  - id: failed_reads_path
+    type: string?
+    inputBinding:
+      prefix: '--failed_out'
+    doc: |
+        specify the file to store reads that cannot pass the filters.
+  - id: read1_adapter_sequence
+    type: string?
+    inputBinding:
+        prefix: '--adapter_sequence'
+    doc: |
+         the adapter for read1. For SE data, if not specified, the adapter will be auto-detected. For PE data, this is used if R1/R2 are found not overlapped.
+  - id: read2_adapter_sequence
+    type: string?
+    inputBinding:
+        prefix: '--adapter_sequence_r2'
+    doc: |
+         the adapter for read2. For PE data, this is used if R1/R2 are found not overlapped.
+  - id: minimum_read_length
+    type: int?
+    inputBinding:
+        prefix: '--length_required'
+    doc: |
+        reads shorter than length_required will be discarded, default is 15.
+
+  - id: json_output_path 
```

Comment:
I would make the `json_output_path` and `html_output_path` inputs to be optional since as you mentioned they are outputted anyway.

Reply:
Or better yet, set their default values to what fastp names them by default. e.g. for `html_output_path` set `default: 'fastp.html'`.

That way the outputs know how to find the files